.github/workflows/node.js.yml

```diff
@@ -6,19 +6,107 @@ name: Node.js CI
 on: [push, pull_request]
 
 jobs:
-  build:
-
+  common:
+    name: Common tests (Node.js v${{ matrix.node-version }})
     runs-on: macos-latest
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
```

Comment:
would it be possible to move these versions into common variables section, we we could bump them for all affected jobs when necessary?

Reply:
same about xcode/device versions

Reply:
not sure if github supports this, or failing that, if it supports yaml aliases.  I'll see what I can do

Reply:
this is currently not possible:

- The [`defaults`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaults) prop should support this sort of thing, but it only currently supports providing defaults for the `run` prop of a job.  The `strategy` prop of a job (which includes the `matrix`) [does not have access](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#context-availability) to any user-configurable Contexts.  I tried to create a JSON environment variable (via the top-level `env` prop) and use the `fromJSON()` function, but again, `strategy.matrix` cannot access the `env` Context.
- YAML anchors are not supported in workflow files, so I can't just use an alias.  Tried it!

I can't think of any other way to do this at the moment.

Reply:
yep, Azure seems to be more advanced from that perspective